hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/table/TestCleaner.java

```diff
@@ -1148,7 +1167,9 @@ public void testKeepLatestCommits(boolean simulateFailureRetry, boolean enableIn
             .withIncrementalCleaningMode(enableIncrementalClean)
             .withFailedWritesCleaningPolicy(HoodieFailedWritesCleaningPolicy.EAGER)
             .withCleanBootstrapBaseFileEnabled(enableBootstrapSourceClean)
-            .withCleanerPolicy(HoodieCleaningPolicy.KEEP_LATEST_COMMITS).retainCommits(2).build())
+            .withCleanerPolicy(HoodieCleaningPolicy.KEEP_LATEST_COMMITS)
+                .retainCommits(2)
```

Comment:
instead of overloading this test, can we write a new one on the `TestCleanPlanExecutor` level? We are trying to discourage overloading functional tests like these for functionality that can be tested more easily (often times, its not possible). but in this case, I think it is.

Reply:
I am fine with adding a new class for testing this. 

`TestCleanPlanExecutor` can have  `testInvalidCleaningTriggerStrategy()`. But adding the happy flow there will lead to code duplication since I need to simulate the commit timeline exactly as is to be able to show the working of this cleaner trigger strategy. Please let me know and I will do the needful. @nsivabalan @vinothchandar

Reply:
yeah. can we try to add more tests around clean plan executor level. at a higher level, we can have just one set of cleaner tests. But for diff strategies would be better to avoid more tests at write client or table level.

Reply:
got it